src/pipeline.ts

```diff
@@ -482,14 +523,400 @@ export async function scanRssFeeds() {
 		});
 		await indexNewTorrents();
 		for (const [i, candidate] of candidatesSinceLastTime.entries()) {
+			const candidateLog = `${chalk.bold.white(candidate.name)} from ${candidate.tracker}`;
 			logger.verbose({
 				label: Label.RSS,
-				message: `Processing release ${i + 1}/${
-					candidatesSinceLastTime.length
-				}`,
+				message: `(${i + 1}/${candidatesSinceLastTime.length}) ${candidateLog}`,
 			});
-			await checkNewCandidateMatch(candidate, Label.RSS);
+			await checkNewCandidateMatch(candidate, candidateLog, Label.RSS);
 		}
 		logger.info({ label: Label.RSS, message: "Scan complete" });
 	}
 }
+
+export async function injectSavedTorrents() {
+	const { flatLinking, injectDir, linkDir, outputDir } = getRuntimeConfig();
+	const targetDir = injectDir ?? outputDir;
+	const targetDirLog = chalk.bold.magenta(targetDir);
+
+	if (injectDir) {
+		// injectDir defined only by `cross-seed inject`
+		logger.warn({
+			label: Label.INJECT,
+			message: `Manually injecting torrents performs minimal filtering which slightly increases chances of false positives, see the docs for more info`,
+		});
+	}
+	if (!fs.existsSync(targetDir) || !fs.statSync(targetDir).isDirectory()) {
+		logger.info({
+			label: Label.INJECT,
+			message: `No directory found at ${targetDirLog} - ensure it exists and is accessible (verify docker volumes if applicable)`,
+		});
+		return;
+	}
+	const dirContents = await findAllTorrentFilesInDir(targetDir);
+	if (dirContents.length === 0) {
+		logger.info({
+			label: Label.INJECT,
+			message: `No torrent files found to inject in ${targetDirLog}`,
+		});
+		return;
+	}
+	logger.info({
+		label: Label.INJECT,
+		message: `Found ${chalk.bold.white(dirContents.length)} torrent file(s) to inject in ${targetDirLog}`,
+	});
+
+	const { searchees } = await findSearchableTorrents(Label.INJECT, {
+		useFilters: false,
+	});
+
+	const toDelete = new Set<string>();
+	const toRecheck = new Set<string>();
+
+	// Usually source got deleted or partial injection never completes
+	function shouldCleanUpTorrent(torrentFilePath: string) {
+		return fs.statSync(torrentFilePath).mtimeMs < Date.now() - ms("1 week");
+	}
```

Comment:
I'm wondering if cleaning up torrents would make more sense as a separate job. What do you think?